.golden/kotlinEnumValueClassSpec/golden

```diff
@@ -0,0 +1,12 @@
+@Parcelize
+@Serializable
+@JvmInline
+value class EnumAsValueClass(val value: String) : Parcelable {
+    companion object {
+        val First = EnumAsValueClass("first")
+        val Second = EnumAsValueClass("second")
+        val Third = EnumAsValueClass("third")
+
+        val entries = listOf(First, Second, Third)
+    }
+}
```

Comment:
Just so I understand, in kotlin you will have something like

```
when enumAsValueClass is
  First -> x
  Second -> y
  Third -> z
  else -> unknownEquivalent
```

You don't really get exhaustiveness on all of these pattern matches which kind of stinks but you prevent hosing the app which seems pretty useful.

Reply:
In the sense that, if someone adds `Fourth` you need to go search out all of these matches

Reply:
Yes, that does suck and it is a tradeoff to consider. I'm simultaneously looking for a way to automatically handle `List<Enum<T>>` with fallbacks in kotlinx.serialization, so this is more of a hedge against the possibility that I can't find a solution there.

Reply:
I think this is fine for now though. It definitely restates the need for the ReaderT to avoid passing these inputs all the way down through Moat every time.